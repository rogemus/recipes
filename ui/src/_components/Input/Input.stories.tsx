import { Input } from "./Input";

import type { Meta, StoryObj } from "@storybook/react";

import { fn } from "@storybook/test";

const meta: Meta<typeof Input> = {
  component: Input,
  title: "Input",
  args: {
    onChange: fn,
    id: "id",
    name: "Email",
    type: "email",
    label: "Email",
    placeholder: "Email ...",
  },
};

export default meta;

type Story = StoryObj<typeof Input>;

export const Default: Story = {};
