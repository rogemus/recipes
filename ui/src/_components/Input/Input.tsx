import type { FC } from "react";
import type { InputProps } from "./Input.types";
import styles from "./Input.module.css";

const Input: FC<InputProps> = ({
  type = "text",
  id,
  placeholder,
  name,
  defaultValue,
  label,
  testId = "Input",
  onChange,
}) => {
  return (
    <div className={styles.wrapper} data-testid={testId}>
      <label
        className={styles.label}
        data-testid={`${testId}-label`}
        htmlFor={id}
      >
        {label}
      </label>
      <input
        data-testid={`${testId}-input`}
        className={styles.input}
        type={type}
        id={id}
        name={name}
        {...(onChange && { onChange })}
        {...(placeholder && { placeholder })}
        {...(defaultValue && { defaultValue })}
      />
    </div>
  );
};

export { Input };
