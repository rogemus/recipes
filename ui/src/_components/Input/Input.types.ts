import { Comp } from "@/_models";
import { ChangeEventHandler } from "react";

export type InputProps = {
  label: string;
  type?: HTMLInputElement["type"];
  placeholder?: HTMLInputElement["placeholder"];
  defaultValue?: string;
  onChange?: ChangeEventHandler<HTMLInputElement>;
  id: string;
  name: string;
} & Comp;
