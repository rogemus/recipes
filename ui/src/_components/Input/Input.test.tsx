import { expect, test, describe } from "vitest";
import { render } from "@testing-library/react";
import { Input } from "./Input";
import { InputProps } from "./Input.types";

const defaultProps: InputProps = {
  label: "test-label",
  id: "test-id",
  name: "test-name",
  placeholder: "Test placeholder",
};

describe("@component/Input", () => {
  test("renders component correctly", () => {
    const { container } = render(<Input {...defaultProps} />);

    expect(container).toMatchSnapshot();
  });
});
