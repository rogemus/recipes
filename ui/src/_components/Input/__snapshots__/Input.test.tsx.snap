// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`@component/Input > renders component correctly 1`] = `
<div>
  <div
    class="_wrapper_838bd5"
    data-testid="Input"
  >
    <label
      class="_label_838bd5"
      data-testid="Input-label"
      for="test-id"
    >
      test-label
    </label>
    <input
      class="_input_838bd5"
      data-testid="Input-input"
      id="test-id"
      name="test-name"
      placeholder="Test placeholder"
      type="text"
    />
  </div>
</div>
`;
